bin/benchmark/Sources/benchmark/Diff/DiffAnalysis.swift

```diff
@@ -0,0 +1,219 @@
+                    return MetricAnalysis(
+                        metricName: afterMetric.displayName,
+                        metricID: afterMetric.id,
+                        change: beforeValues == afterValues ? .same : .differentChecksum,
+                        before: beforeMetric.values.formatted(),
+                        after: afterMetric.values.formatted()
+                    )
+                }
+                
+            case (.checksum(let beforeValues), .checksum(let afterValues)):
+                return MetricAnalysis(
+                    metricName: afterMetric.displayName,
+                    metricID: afterMetric.id,
+                    change: beforeValues[0] == afterValues[0] ? .same : .differentChecksum,
+                    before: beforeMetric.values.formatted(),
+                    after: afterMetric.values.formatted()
+                )
+                
+            default:
+                throw Error.typeMismatchComparingBenchmarkValues(id: afterMetric.id)
+        }
+       
+        var warnings: [String] = []
+        switch beforeMetric.values {
+            case .duration, .bytesInMemory:
+                if !beforeNumbers.looksReasonablyNonBiased() {
+                    warnings.append(Self.inputBiasDescription(metric: beforeMetric, sampleName: "before", numbers: beforeNumbers))
+                }
+                if !afterNumbers.looksReasonablyNonBiased() {
+                    warnings.append(Self.inputBiasDescription(metric: afterMetric, sampleName: "after", numbers: afterNumbers))
+                }
+                    
+            default:break
+        }
+        
+        let change: MetricAnalysis.Change
+        var footnotes: [MetricAnalysis.Footnote]
+        
+        let tTestResult = independentTTest(beforeNumbers, afterNumbers)
+        let footnoteValues: [(String, String)] = [
+            ("t-statistic", footnoteNumberFormatter.string(from: NSNumber(value: tTestResult.tStatistic))!),
+            ("degrees of freedom", tTestResult.degreesOfFreedom.description),
+            ("critical value", footnoteNumberFormatter.string(from: NSNumber(value: tTestResult.criticalValue))!),
+        ]
+        
+        if tTestResult.seriesAreProbablyTheSame {
+            change = .same
+            footnotes = [
+                .init(text: "The before and after values are similar enough that the most probable explanation is that they're random samples from the same data set.", values: footnoteValues)
+            ]
+        } else {
+            let approximateChange = (afterNumbers.mean() - beforeNumbers.mean()) / beforeNumbers.mean()
+            change = .differentNumeric(percentage: approximateChange)
+            
+            footnotes = [
+                .init(text: "The before and after values are different enough that the most probable explanation is that random samples from two different data sets.", values: footnoteValues)
```

Comment:
I think there is a missing word here.

Also, I find this text and the text above a bit hard to follow, not having a statistics background.

Is it correct to simply say:
- ✅ No statistically significant change
- ⚠️ Possible regression

And display that in the table, so that you can see at a glance what might've regressed?

Reply:
Franklin and I talked about this offline. I feel that the diff tool should avoid trying to draw conclusions and instead aim to present the necessary information that a human needs to consider when analyzing the performance differences.

To that end, I feel like we should be careful to avoid present inconclusive results that may be incorrectly interpreted as absolute success/failure values. Still, we could use colors or formatting to guide the person who looks at the results to the values that they need to focus their attention on.

Reply:
I addressed this in c34eaaa078ea63733969ca2e21f5449bf7d336ef by adding color as follows:
- green: statistically significant improvement in performance _and_ no signs of bias in these measurements
- red: statistically significant regression in performance _and_ signs of bias in these measurements
- yellow: possible signs of signs of bias in either the before or after measurements